### PR TITLE
Fix MSVC compiler warnings

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
* reg. conversion from 'size_t' to 'int', possible loss of data (C4267)
* reg. declaration hides class member (C4458)
* reg. local variable is initialized but not referenced (C4189)
* reg. declaration hides function parameter (C4457)

### CODE REVIEW

File: src/notation/view/widgets/editstyle.cpp [hunk 1]
```diff
@@ -862,8 +862,8 @@ EditStyle::EditStyle(QWidget* parent)
             connect(checkBox, &QCheckBox::stateChanged, setSignalMapper, mapFunction);
         } else if (auto button = qobject_cast<QAbstractButton*>(sw.widget)) {
             connect(button, &QAbstractButton::toggled, setSignalMapper, mapFunction);
-        } else if (auto groupBox = qobject_cast<QGroupBox*>(sw.widget)) {
-            connect(groupBox, &QGroupBox::toggled, setSignalMapper, mapFunction);
+        } else if (auto myGroupBox = qobject_cast<QGroupBox*>(sw.widget)) {
```
Comment (alexpavlov96) [inline src/notation/view/widgets/editstyle.cpp:865]: why is it needed? didn't see the same name above

File: src/engraving/compat/midi/event.h [hunk 2]
```diff
@@ -102,8 +102,8 @@ class NPlayEvent : public PlayEvent
 
     const Note* _note = nullptr;
     const Harmony* _harmony = nullptr;
-    int _origin = -1;
-    int _discard = 0;
+    size_t _origin = (size_t)-1;
```
Comment (alexpavlov96) [inline src/engraving/compat/midi/event.h:105]: (size_t)-1 is safe, but it's rather std::numeric_limits<size_t>::max()
but then we have 'getOriginatingStaff() + 1' and it's confusing, even though will give us zero
maybe leave int everywhere?

File: src/notation/view/widgets/editstyle.cpp (hunk 1 above)
Comment (Jojo-Schmitz) [reply src/notation/view/widgets/editstyle.cpp:865]: It is there, somewhere, check the output of other Windowes builds:
```
2023-05-09T13:35:01.2361479Z D:/a/MuseScore/MuseScore/src/notation/view/widgets/editstyle.cpp(865): warning C4458: declaration of 'groupBox' hides class member
2023-05-09T13:35:01.4484152Z D:\a\MuseScore\MuseScore\build.release\src\notation\ui_editstyle.h(1017): note: see declaration of 'Ui_EditStyleBase::groupBox'
```

File: src/engraving/compat/midi/event.h (hunk 2 above)
Comment (Jojo-Schmitz) [reply src/engraving/compat/midi/event.h:105]: It was `int` everywhere. And lead to a warning, I beliebe it is this'
```
2023-05-09T13:31:26.9125539Z D:/a/MuseScore/MuseScore/src/importexport/midi/internal/midiexport/exportmidi.cpp(317): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data
```

File: src/notation/view/widgets/editstyle.cpp (hunk 1 above)
Comment (Jojo-Schmitz) [reply src/notation/view/widgets/editstyle.cpp:865]: Maybe we can fix it there, will have a look.

File: src/engraving/compat/midi/event.h (hunk 2 above)
Comment (Jojo-Schmitz) [reply src/engraving/compat/midi/event.h]: But we can fix it with a single cast in exportmidi.cpp
I'll do that

**Review by RomanPudashkin — APPROVED** [review]

**Pull request merged**